apps/condo/domains/meter/hooks/useUpdateMeterModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { Typography } from 'antd'
+import get from 'lodash/get'
+import pick from 'lodash/pick'
+import { useIntl } from '@core/next/intl'
+
+import { Button } from '@condo/domains/common/components/Button'
+
+import { BaseMeterModalForm } from '../components/BaseMeterModal/BaseMeterModalForm'
+import { Meter } from '../utils/clientSchema'
+import { IMeterUIState } from '../utils/clientSchema/Meter'
+
+const DeleteMeterButton = ({ meter, updateMeterAction }) => {
+    const intl = useIntl()
+    const DeleteMessage = intl.formatMessage({ id: 'Delete' })
+
+    const handleDeleteButtonClick = useCallback(() => updateMeterAction({ deletedAt: new Date().toDateString() }, meter),
+        [meter, updateMeterAction])
+
+    return (
+        <Button
+            type={'text'}
+            danger
+            onClick={handleDeleteButtonClick}
+        >
+            {DeleteMessage}
+        </Button>
+    )
+}
+
+const INITIAL_VALUES_KEYS = [
+    'accountNumber', 'number', 'resource', 'place', 'numberOfTariffs', 'installationDate',
+    'commissioningDate', 'sealingDate', 'verificationDate', 'nextVerificationDate', 'controlReadingsDate',
+]
+
+export const useUpdateMeterModal = (refetch) => {
+    const intl = useIntl()
+    const MeterNumberMessage = intl.formatMessage({ id: 'pages.condo.meter.NumberOfMeter' })
+
+    const [selectedMeter, setSelectedMeter] = useState<IMeterUIState>()
+    const meterNumber = get(selectedMeter, 'number')
+
+    const updateMeterAction = Meter.useUpdate({}, () => {
+        setSelectedMeter(null)
+        refetch()
+    })
+
+    const initialValues = useMemo(() => {
+        if (selectedMeter) {
+            return pick(selectedMeter, INITIAL_VALUES_KEYS)
+        }
+    }, [selectedMeter])
```

Comment:
is it ok to be `undefined` here if there is no `selectedMeter`?

Reply:
yes, if `selectedMeter` is undefined then update modal will not be shown